### ISSUE TITLE
Add loss functions

### ISSUE THREAD

**Issue by vadixidav** [vadixidav/mli#4]
It is necessary to have the capability to define loss functions. At the very least, basic primitives need to exist to define the loss functions. For L2 distance, that might include subtraction, squaring, and average pooling. We should also have a cross-entropy loss for classification and the primitives necessary to create that from scratch.
